### PR TITLE
fix(oneinch): fail closed on a min-return bind mismatch for known selectors

### LINKED ISSUE #1438: P2: 1inch/kyber/li.fi swaps have no sign-time min-out bind (aggregator calldata trusted verbatim)
## P2 - 1inch / kyber / li.fi swaps have no sign-time min-out bind (aggregator-trusted)

Found in a fund-safety spike. **Cross-consumer**: this is the shared SDK general-swap build, so it affects flagship ios/android/windows swaps, not just the agent app - which is exactly why this is a file-and-decide, not a rushed patch.

### The gap
- **What's signed** (`packages/core/mpc/keysign/swap/build.ts`, the `evm` arm of `txMsg`): the aggregator's **opaque calldata `data` is forwarded verbatim** into the signed OneInchTransaction. The min-out / `amountOutMin` floor lives *inside* that calldata (encoded by the router per the requested slippage).
- **Displayed out-amount is a separate, unbound field**: `toAmountDecimal: fromChainAmountDisplay(quote.dstAmount, ...)` / `OneInchQuoteSchema { dstAmount, tx }` - `dstAmount` is the quote's expected-out, **never cross-checked against the floor baked into `data`**.
- **No decode/assert anywhere**: grep for `decodeFunctionData|minReturn|amountOutMin|amountOutMinimum|assert.*dstAmount|verifyCalldata` across `packages/core/chain/swap`, `packages/core/mpc/keysign/swap`, `packages/sdk/src/tools/swap` is **empty**. Slippage is only a *request param* at quote-time (`kyber/api/route.ts:52,62`); the returned calldata is trusted.
- **No price-impact ceiling either**: `assertJupiterPriceImpactWithinCeiling` is wired for **Jupiter only** (`jupiter/api/getJupiterSwapQuote.ts:124`); 1inch/kyber/li.fi expose no `priceImpactPct` and get no equivalent.

### Failure scenario
A compromised aggregator response or swap-proxy returns `{ dstAmount: "1000000000" (~1000 USDC, DISPLAYED), tx: { data: <router calldata with amountOutMin = 1> } }` -> card shows "receive ~1000 USDC", the vault signs a swap that accepts **1 wei out** -> sandwich-to-zero. WYSIWYS is not cryptographically enforced for these routes.

### Severity: P2 (trust-dependency / defense-in-depth)
Under an honest aggregator + honest first-party proxy the min-out is correct, so this is not a spontaneous drain - it's the *same threat model* that justified the Jupiter price-impact guard (SOL-02), left open for the EVM aggregators. **Not P0/P1** because it requires a compromised/malicious aggregator or proxy.

### Already-protected (not affected)
- **CowSwap**: encodes explicit `buyAmount` (the order floor) into the signed EIP-712 digest via `encodeCowSwapKeysignData` - floor is *in the signed bytes*.
- **SwapKit BTC**: `verifySwapKitBitcoinPsbtOutputs` binds to/amount.
- **Jupiter**: price-impact ceiling enforced.
The gap is specifically **1inch + kyber + li.fi**.

### Fix options (needs an SDK-owner decision - cross-consumer)
1. **Proper fix**: decode each router's `minReturn`/`amountOutMin` from the calldata and assert `>= dstAmount * (1 - maxSlippage)` at sign-time. Correct + cryptographic, but needs per-router ABI decoders (1inch V5/V6, Kyber MetaAggregationRouter, LiFi Diamond) - non-trivial, which is likely why it wasn't done.
2. **Interim (lower-risk, possibly agent/app-layer)**: bind/echo the requested `slippage` + `dstAmount` into the consent card so the user SEES the floor they're agreeing to, and formally document the aggregator-proxy as a hardened trust boundary. Doesn't cryptographically bind but closes the WYSIWYS-display half without a deep shared-SDK change.
3. Accept the aggregator-proxy as the trust boundary + document it explicitly.

Recommend option 2 as the near-term (cross-consumer-safe) and option 1 as the tracked proper fix. Note for agents: reproduce by tracing `build.ts` evm arm -> confirm `data` is signed verbatim with no minReturn decode.

### PR DESCRIPTION
**Pull request by gomesalexandre**
Closes vultisig/vultisig-sdk#1438

Added a fail-closed 1inch known-selector min-out bind: decode V5/V6 `swap` / `unoswap*` / `ethUnoswap*` calldata and refuse when `minReturn < dstAmount * (1 - slippage)`. Unknown selectors stay signable so a router upgrade does not brick honest swaps. Wired at `getOneInchSwapQuote` after the existing router + tx.value guards. Blast radius: 1inch quote construction only; Kyber/LiFi calldata is still opaque (per-router ABI decode is a follow-up). Not verified: a live 1inch quote + signed swap.

## what I ran
```
yarn test:core packages/core/chain/swap/general/oneInch/decodeMinReturn.test.ts packages/core/chain/swap/general/oneInch/api/getOneInchSwapQuote.test.ts
 Test Files  2 passed (2)
      Tests  20 passed (20)
   Duration  517ms

yarn tsc --project .config/tsconfig.json --noEmit --pretty false
(exit 0)
```

closes vultisig/vultisig-sdk#1438

## Summary by CodeRabbit

- **New Features**
  - Added validation for 1inch swap quotes to ensure minimum received amounts meet the requested slippage tolerance.
  - Supports recognized 1inch swap transaction formats while allowing unknown or undecodable calldata to proceed.

- **Bug Fixes**
  - Prevents signing known 1inch swaps when their minimum output falls below the acceptable slippage-adjusted amount.

- **Tests**
  - Added coverage for supported transaction formats, invalid data, boundary values, and rejected minimum outputs.

### CODE REVIEW

**Review by gomesalexandre — COMMENTED** [review]
blocking on current head `8f2464ca8e01d958f1dfd02ebc1061d5396cb4e8`.

`packages/core/chain/swap/general/oneInch/decodeMinReturn.ts` adds a new production module, but `packages/core/chain/package.json` on this head still exports `./swap/general/oneInch/api/getOneInchSwapQuote` and does not export `./swap/general/oneInch/decodeMinReturn`.

That leaves the shared package export map stale. The repo's static quality gate catches it as `packages/core/chain/package.json has stale generated exports` with missing export `./swap/general/oneInch/decodeMinReturn`, which is why current-head `Lint and Type Check` is red.

Please regenerate the shared package exports for this new module and rerun the quality gate before merge.

**Comment by gomesalexandre** [comment]
pushed `95737394689c11e6ab3902cb4e20f9725c125e58` to close the stale export-map blocker on this head.

what changed
- added the missing `./swap/general/oneInch/decodeMinReturn` export entry to `packages/core/chain/package.json`
- that matches the new production module already present at `packages/core/chain/swap/general/oneInch/decodeMinReturn.ts`
- this is the exact package-export hole called out in the prior current-head review

receipts
- exact head now exposes the missing export path in `packages/core/chain/package.json`
- fresh GitHub CI restarted on `95737394689c11e6ab3902cb4e20f9725c125e58` and is re-running now
- no local exact-head rerun here because the host is critically disk-bound (`No space left on device` while preparing a disposable clone), so i'm leaving the validation truth to current-head CI on this follow-up

not approving. this branch still contains gomes-authored commits and needs a non-gomes current-head review after CI goes green.

File: packages/core/chain/swap/general/oneInch/decodeMinReturn.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  },
+  {
+    abi: parseAbi(['function ethUnoswap2(uint256 minReturn, uint256 dex, uint256 dex2)']) as Abi,
+    pick: args => (typeof args[0] === 'bigint' ? args[0] : undefined),
+  },
+  {
+    abi: parseAbi(['function ethUnoswap3(uint256 minReturn, uint256 dex, uint256 dex2, uint256 dex3)']) as Abi,
+    pick: args => (typeof args[0] === 'bigint' ? args[0] : undefined),
+  },
+]
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply packages/core/chain/swap/general/oneInch/decodeMinReturn.ts]: pushed `e4a04ad6d` to cover the current-head gap. `decodeMinReturn.ts` now decodes recipient-bearing `unoswapTo*` and `ethUnoswapTo*` selectors, and the focused tests now pin both decode and below-floor refusal paths for those variants. host exact-head rerun is still blocked by local ENOSPC, so current-head CI is the fresh authoritative receipt.

**Comment by gomesalexandre** [comment]
pushed `e4a04ad6d` on top of `957373946` to close the current-head 1inch min-return decoder gap.

what changed
- `packages/core/chain/swap/general/oneInch/decodeMinReturn.ts` now decodes recipient-bearing `unoswapTo`, `unoswapTo2`, `unoswapTo3`, `ethUnoswapTo`, `ethUnoswapTo2`, and `ethUnoswapTo3` selectors, so those calldata shapes no longer bypass the known-selector min-out floor
- the new focused tests in `decodeMinReturn.test.ts` cover both successful minReturn extraction and fail-closed low-minimum refusal for the recipient-bearing variants
- folded the local camelCase cleanup for the newly introduced decoder/test constants while touching the files

receipts
- host exact-head rerun is blocked right now by local `ENOSPC` during disposable clone/worktree prep, so I am not claiming a fresh local green receipt on this pushed head
- current-head CI should rerun on `e4a04ad6d` and is the authoritative fresh validation for this follow-up

not approving. this branch still contains gomes-authored commits and needs a non-gomes current-head review.

**Comment by gomesalexandre** [comment]
pushed `dd334fd7e` on top of `e4a04ad6d` to close the current-head 1inch selector gap.

what changed
- `packages/core/chain/swap/general/oneInch/decodeMinReturn.ts` now decodes all six recipient-bearing router methods with `uint256 to`, matching the real AggregationRouterV6 selectors instead of the wrong `address to` signatures
- `packages/core/chain/swap/general/oneInch/decodeMinReturn.test.ts` now encodes those recipient-bearing fixtures with bigint `to` values, so the regression exercises the real selectors
- `packages/core/chain/swap/general/oneInch/api/getOneInchSwapQuote.test.ts` got the stale `REAL_ROUTER` reference cleanup that typecheck exposed while re-grounding the fix

exact-head receipts on `dd334fd7e`
- `yarn vitest run packages/core/chain/swap/general/oneInch/decodeMinReturn.test.ts packages/core/chain/swap/general/oneInch/api/getOneInchSwapQuote.test.ts` -> `24 passed`
- `yarn tsc --project .config/tsconfig.json --noEmit`
- `git diff --check` ✅

not approving. this branch is gomes-authored and still needs external current-head review for four-eyes.

**Review by gomesalexandre — COMMENTED** [review]
reviewed current head `dd334fd7e10a3545c104e005af1b3c530e627f20`.

looks sound on the current head. no new blocker surfaced.

what i re-grounded
- `packages/core/chain/package.json` now exports `./swap/general/oneInch/decodeMinReturn`, so the earlier stale export-map blocker is closed
- the current head now covers the recipient-bearing 1inch router variants and the follow-up notes pin the real selector fix to `uint256 to`, which was the substantive gap on the intermediate head
- current-head github checks are green across unit, vitest integration, node storage, lint/typecheck, and test summary
- the current thread already carries exact-head receipts for `dd334fd7e`: focused `vitest` on the 1inch decoder + quote tests, `tsc --noEmit`, and `git diff --check`

not approving. this branch is gomes-authored and still needs a non-gomes current-head review for four-eyes.

**Review by rcoderdev — APPROVED** [review]

**Comment by rcoderdev** [comment]
Closing this because Closing this version because it checks known 1inch selectors during quote construction, while #1438 calls for a sign-time minimum-output guard across 1inch, Kyber, and LI.FI. Finishing this branch requires a broader integration and fresh multi-provider QA; the safety requirement remains open.

This PR has been open for more than two weeks, and we do not plan to carry this version forward.

If the change is still needed, please open a fresh PR against current main with the updated context.
